3-Authorization-II/2-call-api-b2c/API/controllers/todolist.js

```diff
@@ -0,0 +1,84 @@
+const lowdb = require('lowdb');
+const FileSync = require('lowdb/adapters/FileSync');
+const adapter = new FileSync('./data/db.json');
+const db = lowdb(adapter);
+const { hasRequiredDelegatedPermissions } = require('../auth/permissionUtils');
+
+const authConfig = require('../authConfig');
+
+exports.getTodo = (req, res, next) => {
+    if (hasRequiredDelegatedPermissions(req.authInfo, authConfig.protectedRoutes.todolist.delegatedPermissions.read)) {
+        try {
+            const id = req.params.id;
+            const todo = db.get('todos').find({ id: id }).value();
+            res.status(200).send(todo);
+        } catch (error) {
+            next(error);
+        }
+    } else {
+        next(new Error('User does not have the required permissions'));
+    }
+};
+
+exports.getTodos = (req, res, next) => {
+    if (hasRequiredDelegatedPermissions(req.authInfo, authConfig.protectedRoutes.todolist.delegatedPermissions.read)) {
+        try {
+            const owner = req.authInfo['oid'];
```

Comment:
```suggestion
            const owner = req.authInfo['oid'];
```

Should use `sub` claim instead